### ISSUE TITLE
Challenge 1

### ISSUE THREAD

**Issue by RidhikaJoshi** [scaleracademy/scaler-september-open-source-challenge#368]
name: Ridhika Joshi
github_user_name: RidhikaJoshi

**Comment by RidhikaJoshi** [comment]
![image](https://user-images.githubusercontent.com/92508014/188269288-38841ef4-bfcd-4faa-be59-13ca54dcfca2.png)

**Comment by RidhikaJoshi** [comment]
<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/188321693-b07d9734-b13a-430b-8875-1124e0117b16.png">

**Comment by RidhikaJoshi** [comment]
![image](https://user-images.githubusercontent.com/92508014/188478634-32e6baf8-a094-4b3c-9000-21768c3c0b33.png)

**Comment by RidhikaJoshi** [comment]
https://github.com/scaleracademy/scaler-september-open-source-challenge/pull/718/commits
![image](https://user-images.githubusercontent.com/92508014/188638809-90e8fae4-8871-43d4-ab00-ecab53a9108d.png)

**Comment by RidhikaJoshi** [comment]
Before reset
<img width="960" alt="Screenshot 2022-09-09 215010" src="https://user-images.githubusercontent.com/92508014/189397304-b6038b94-92f0-4453-8189-6b3a63d6af26.png">
After reset
<img width="960" alt="Screenshot 2022-09-09 215615" src="https://user-images.githubusercontent.com/92508014/189397348-f61183a8-56fb-40c2-8fae-822020295498.png">

**Comment by RidhikaJoshi** [comment]
Challenge 10
<img width="646" alt="Screenshot 2022-09-10 152318" src="https://user-images.githubusercontent.com/92508014/189478551-5e2e9a08-f4f1-448e-b7f0-ceaedcfc0302.png">
<img width="960" alt="Screenshot 2022-09-10 153358" src="https://user-images.githubusercontent.com/92508014/189478555-583fbf48-399e-4f3d-b5ef-170b83b44d93.png">

**Comment by RidhikaJoshi** [comment]
Challenge-11
<img width="1021" alt="Challenge 11" src="https://user-images.githubusercontent.com/92508014/189529281-4a50eb84-7ce5-415a-9e92-6e2254ae756b.png">

**Comment by RidhikaJoshi** [comment]
Challenge 12
<img width="960" alt="12" src="https://user-images.githubusercontent.com/92508014/189707636-376b4644-6d4e-4150-8627-c9493b3cfca6.png">

**Comment by RidhikaJoshi** [comment]
<img width="960" alt="1" src="https://user-images.githubusercontent.com/92508014/190178581-9b220fdb-b327-44d6-98bb-dd5321d96579.png">
<img width="960" alt="2" src="https://user-images.githubusercontent.com/92508014/190178637-7102299e-98c0-44ba-8036-03ef5ffac491.png">
<img width="960" alt="3" src="https://user-images.githubusercontent.com/92508014/190178663-0746730e-0e04-4ae7-bfe4-7d7cf23c7547.png">
<img width="960" alt="4" src="https://user-images.githubusercontent.com/92508014/190178695-de673e95-776a-4d08-a437-6036cb4bdb90.png">

**Comment by RidhikaJoshi** [comment]
Challenge -15
<img width="960" alt="Screenshot 2022-09-15 192008" src="https://user-images.githubusercontent.com/92508014/190421902-81872e52-efb9-4c25-8e0c-9b90eb1b4890.png">

**Comment by RidhikaJoshi** [comment]
Challenge-16: Fixed Merge Conflict.
Before:
<img width="669" alt="Screenshot 2022-09-16 181344" src="https://user-images.githubusercontent.com/92508014/190643884-433a8adc-0106-45a1-a9c1-48a944bfc476.png">
<img width="960" alt="1" src="https://user-images.githubusercontent.com/92508014/190643952-d20ea023-e52e-42e7-a5f8-e9c0b954604b.png">
<img width="960" alt="2" src="https://user-images.githubusercontent.com/92508014/190643979-828127e3-e387-4e03-b3e9-67e9ee7170d9.png">
After:
<img width="644" alt="Screenshot 2022-09-16 182414" src="https://user-images.githubusercontent.com/92508014/190644043-0dd05516-229d-4abd-8417-36d6fec6a591.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge -17</h1>

<h3>Installed GitHub Desktop and Created Challenge-17 Repository</h3>
<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/190861209-640faf9e-f037-45e5-bf81-4191a7780ece.png">

<h3>Adding Commits</h3>
<img width="960" alt="2" src="https://user-images.githubusercontent.com/92508014/190861437-e8bfca1d-5801-475b-aa4c-bf49dff87523.png">

<h3>Pushing Changes to GitHub Repository</h3>

Repo Link- https://github.com/RidhikaJoshi/Challenge-17

<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/190861569-9850634d-8650-42ab-9140-a6d549a65335.png">

<h3>Creating New Branch(Test-Branch)</h3>
<img width="428" alt="image" src="https://user-images.githubusercontent.com/92508014/190862175-860493f0-8176-42d4-8097-a1ef2d3a2519.png">

<h3>Publishing New Branch to the GitHub</h3>
<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/190862253-2a19a0f6-593f-4370-b899-f373948d8fd3.png">

<h3>Commiting Changes in new branch(Test-Branch)</h3> 
<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/190862431-ea87bffe-13a1-4a49-8af9-a44addbd9c11.png">
<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/190862529-17f668f3-a16b-471d-ac96-ab59e9c66f55.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-18</h3>
<img width="960" alt="1" src="https://user-images.githubusercontent.com/92508014/190903909-4100054c-354b-48c4-b890-bf308203d6ea.png">
<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/190904060-d78a468d-e665-4c7e-9691-5e3cf4054d94.png">
<img width="960" alt="3" src="https://user-images.githubusercontent.com/92508014/190904073-3f28ea03-4677-4514-a99a-333a1799f667.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-19</h1>

<h2>Fixed Lint Errors</h2>
<img width="960" alt="1" src="https://user-images.githubusercontent.com/92508014/191043162-17cc839a-61cd-420d-9ce9-f082c4aa92e2.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-20</h1>
<h2>Created a GitHub Repository</h2>

Repo Link- https://github.com/RidhikaJoshi/Algorithms-Project

<img width="960" alt="1" src="https://user-images.githubusercontent.com/92508014/191316339-3d7ec87a-2879-48a6-af8a-0498480355ec.png">
<h2>Created a project board with a TODO column, In progress, Done and Created a card for all the items in the TODO column</h2>
<img width="960" alt="2" src="https://user-images.githubusercontent.com/92508014/191316414-6bfeb94d-3182-4eba-86df-1ff123af7330.png">

<img width="960" alt="3" src="https://user-images.githubusercontent.com/92508014/191316565-e6c36fee-d721-48b5-8ab2-158236b4ff1c.png">
<img width="960" alt="4" src="https://user-images.githubusercontent.com/92508014/191316587-ca283328-ff9d-43bf-9a78-e0feb2775251.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-21</h1>
I've gone through the Git Alias documentation.
Created 4 alias for git commands(checkout, branch, commit and status)
Screenshot of the alias I've created.


<img width="338" alt="1" src="https://user-images.githubusercontent.com/92508014/191525826-a9b57ecb-cbca-4ded-9c31-1275362aa10c.png">

<img width="323" alt="2" src="https://user-images.githubusercontent.com/92508014/191525853-b567c560-722c-43ad-9c2c-b476ef5874af.png">

<img width="385" alt="3" src="https://user-images.githubusercontent.com/92508014/191525901-aaee8471-8096-4f82-96db-45f9e8bca947.png">

![image](https://user-images.githubusercontent.com/92508014/191526695-973e7224-4f27-4da4-8f37-797bf48f4f19.png)

**Comment by kuraakhilesh8230** [comment]
Do challenge 20 correctly and attach screen shots of all the files of your project which should show done after moving from inprogress

**Comment by kuraakhilesh8230** [comment]
Congrats you have successfully completed milestone 2 and you are good to move onto next milestone
All the best✅️

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-22</h1>

Created Repository- https://github.com/RidhikaJoshi/RidhikaJoshi.github.io 

![image](https://user-images.githubusercontent.com/92508014/191732298-9452b3a3-f80e-4c06-98ff-b11078edcfb3.png)



Uploaded On GitHub Pages - https://ridhikajoshi.github.io/


<img width="960" alt="2" src="https://user-images.githubusercontent.com/92508014/191732204-6f2f1a56-e23a-4671-8ea8-4897e80d2509.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-23</h1>

<h2>Git Branches</h2>

<h3>Branch Categories</h3>

- Regular Git Branches: Development (dev), Master (master) and QA (QA)
- Temporary Git Branches: Bug Fix, Hot Fix, Feature Branches, Experimental Branches and WIP branches.

<h3>Git Branching Naming Convention</h3>

- Start branch name with a Group word.
- Use Unique ID in branch names.
- Use Hyphen or Slash as Separators.
- Git Branch with Author Name.
- Avoid using numbers only.
- Avoid using all naming convention simultaneously.
- Avoid long descriptive names for long-lived branches.

<h2>Pull Request</h2>

<h3>Useful Tips</h3>

- Give your feature branch a clear name.

      A well-formed PR should start with a well-named feature branch.

- Give your commits and PRs active and descriptive titles.

      For both commit messages and PR titles, it's a good idea to use the present imperative tense — for example, use “Fix 
      dashboard typo” instead of “Fixed” or “Fixes.”

- Streamline your process of creating a new PR.

- Give your PR a meaningful description

       A PR description section is where you let the reviewer know why you opened the pull request; the more information you 
       give them before they look at your actual code, the easier their job will be.
       A basic description usually has these elements: Why How and What

- Show your functionality visually, whenever possible

- Review your own PR before you assign it to others

       - Make sure you are caught up with the latest changes in the develop branch; resolve any conflicts that might exist, so 
           that  merging is as easy as pressing a button.
       - Run your tests!
       - Lint your code!

<h2>Contributing Guidlines</h2>

<h3>Finding an Issue to Work On:-</h3>

- Issues available for community contribution:

     help wanted: Open to participation from the community but not necessarily beginner-friendly
     good first issue: Open to participation from the community and friendly towards new contributors.

- Issues not available for community contribution:

     🔒 staff only: Requires infrastructure access or institutional knowledge that would be impractical to provide to the 
      community.

- Issues not ready for work:
 
     🚧 status: blocked: Blocked by other work that needs to be done first
     🧹 status: ticket work required: Needs additional work before it is ready to be taken up
     🚦 status: awaiting triage: Has not been triaged by a maintainer.

- Issues without any of the above labels.

<h3>Contribution Process</h3>

- Identify issue to work on basis on above guidelines, and comment on it to avoid conflicts.
- Write code and submit PR. PR should follow [guidleines](https://opensource.creativecommons.org/contributing-code/pr-guidelines/).
- Wait for code review and address any issues raised as soon as you can.


<h3>Submitting a New Issue</h3>

- Create a new GitHub issue associated with the relevant repository and propose your change there.
- Wait for a project maintainer to evaluate your issue and decide whether it's something that we will accept a pull request for.
- Once the project maintainer has approved the issue (by removing the 🚦 status: awaiting triage label), start work on code as described in above section.

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-24</h1>

Found an issue which is related to my technical skills and I have commented on it if I can contribute:

https://github.com/OpenGenus/cosmos/issues/6341#issuecomment-1256992191

<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/192105761-d42004f9-b95a-408a-95d2-831de8c3c5bd.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-25</h1>

Currently working on the issue (Kadanes Algorithm in Cpp) found yesterday but this issue is not assigned to me and exploring other issues which matches my level.

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-26</h1>

Today I learned about what is hacktoberfest exactly is and I am going to register in that for sure once timer stops.
Joined hacktoberfest discord server.
Still the previous issue is not assigned to me .

Asked the maintainer to assign me this issue.
<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/192325681-b4cb4ae7-9f47-4e6b-9e83-a677352391de.png">

**Comment by kuraakhilesh8230** [comment]
congrats you have completed milestone-3 successfully and you are good to go🎉

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-27</h1>

I have been assigned this issue-https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/issues/181

I registered as a contributor in the hactoberfest

Tips to be kept in mind as a contributor for hactoberfest --

<li>Find out the repositories with hacktoberfest tag and contribute in that.</li>
<li>Make 4 pull/merge requests accepted in the month of October to complete Hacktoberfest.  </li>
<li>PRs must be made to a public, unarchived repository.</li>
<li> If all goes good and all four of your pull requests are approved, you will receive an email from the Hacktoberfest 
     organisers with instructions on how to collect your reward after the 14-day review period. </li>
<li> It is the correct time now to start searching for the repositories with hacktober tag.  </li>

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-28</h1>

Created a pull request for the issue- https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/issues/181

Pull request link-https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/pull/197

<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/192799512-d2336c49-3bc2-4acb-ae21-0b1236662c4d.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-29</h1>

PR got merged- https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/pull/197

<img width="960" alt="image" src="https://user-images.githubusercontent.com/92508014/193056388-79237a10-110c-4cc4-9d74-b8eef3c7310a.png">


<h2>Some Upvoted Ideas</h2>

<img width="629" alt="1" src="https://user-images.githubusercontent.com/92508014/193056665-9e73e6db-094b-4bfb-8458-a6480c89eb75.png">

<img width="653" alt="2" src="https://user-images.githubusercontent.com/92508014/193056699-cd4d2a1c-d7da-4c84-a483-447bc3865e12.png">

**Comment by RidhikaJoshi** [comment]
<h1>Challenge-30</h1>

Successfully completed all the challenges. Thanks to Scaler for organizing it, everyone in the community was helpful and encouraged a lot during these 30 days. It was a great learning experience with Scaler Open-Source. As I was a beginner to GitHub so I didn't have even the basic knowledge of using GitHub, but due to these challenges only I was able to learn and understand many new and important things.
Lastly,I will continue my journey of open source, really excited about Hactoberfest as this will be the first time I will participate.

**Comment by Mrjoy832** [comment]
## Milestone-4 done ✅
